src/modules/stakewise/stakewise-daemon/server/status/get-validator-statuses.go

```diff
@@ -0,0 +1,169 @@
+package swstatus
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/nodeset-org/eth-utils/beacon"
+	swapi "github.com/nodeset-org/hyperdrive/modules/stakewise/shared/api"
+	"github.com/nodeset-org/hyperdrive/shared/types"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/gorilla/mux"
+	"github.com/nodeset-org/hyperdrive/daemon-utils/server"
+)
+
+// ===============
+// === Factory ===
+// ===============
+
+type statusGetValidatorsStatusesContextFactory struct {
+	handler *StatusHandler
+}
+
+func (f *statusGetValidatorsStatusesContextFactory) Create(args url.Values) (*statusGetValidatorsStatusesContext, error) {
+	c := &statusGetValidatorsStatusesContext{
+		handler: f.handler,
+	}
+	inputErrs := []error{}
+	return c, errors.Join(inputErrs...)
+}
+
+func (f *statusGetValidatorsStatusesContextFactory) RegisterRoute(router *mux.Router) {
+	server.RegisterQuerylessGet[*statusGetValidatorsStatusesContext, swapi.ValidatorStatusData](
+		router, "status", f, f.handler.serviceProvider.ServiceProvider,
+	)
+}
+
+// ===============
+// === Context ===
+// ===============
+
+type statusGetValidatorsStatusesContext struct {
+	handler *StatusHandler
+}
+
+func (c *statusGetValidatorsStatusesContext) PrepareData(data *swapi.ValidatorStatusData, opts *bind.TransactOpts) error {
+	sp := c.handler.serviceProvider
+	bc := sp.GetBeaconClient()
+	w := sp.GetWallet()
+	nc := sp.GetNodesetClient()
+	registeredPubkeys, err := nc.GetRegisteredValidators()
+	if err != nil {
+		return fmt.Errorf("error getting registered validators: %w", err)
+	}
+	privateKeys, err := w.GetAllPrivateKeys()
+	if err != nil {
+		return fmt.Errorf("error getting private keys: %w", err)
+	}
+	publicKeys, err := w.DerivePubKeys(privateKeys)
+	if err != nil {
+		return fmt.Errorf("error getting public keys: %w", err)
+	}
+	statuses, err := bc.GetValidatorStatuses(context.Background(), publicKeys, nil)
+	if err != nil {
+		return fmt.Errorf("error getting validator statuses: %w", err)
+	}
+	var generatedValidators, uploadedNodesetValidators, uploadedStakewiseValidators, registeredStakewiseValidators, withdrawalDone, withdrawalPossible, exitedSlashed, exitedUnslashed, activeSlashed, activeExited, activeOngoing, pendingQueued, pendingInitialized []beacon.ValidatorPubkey
+
+	for _, pubKey := range publicKeys {
+		if IsWithdrawalDone(pubKey, statuses) {
```

Comment:
I'd replace all of these with a simple `switch` on the Pubkey status and remove the `Is...` functions below. In fact I'd probably just return the Beacon status itself instead of making individual slices for each status type, which will greatly simplify the response type.

Reply:
done